app/scripts/controllers/transactions/index.js

```diff
@@ -193,12 +193,12 @@ class TransactionController extends EventEmitter {
     }
     txUtils.validateTxParams(normalizedTxParams)
     // construct txMeta
-    const { transactionCategory, getCodeResponse } = await this._determineTransactionCategory(txParams)
     let txMeta = this.txStateManager.generateTxMeta({
       txParams: normalizedTxParams,
       type: TRANSACTION_TYPE_STANDARD,
-      transactionCategory,
     })
+    const { transactionCategory, getCodeResponse } = await this._determineTransactionCategory(txParams)
+    txMeta.transactionCategory = transactionCategory
```

Comment:
Similarly, what is the difference between passing the `transactionCategory` to the method that generates the txMeta and setting the property on the `txMeta` directly?

Reply:
see https://github.com/MetaMask/metamask-extension/pull/7473/files#r348870296

no difference, but we can only get `transactionCategory` from the async call, which needs to happen after `generateTxMeta`